cmake/macros/macro_deal_ii_pickup_tests.cmake

```diff
@@ -183,7 +183,15 @@ MACRO(DEAL_II_PICKUP_TESTS)
   ENABLE_TESTING()
 
   SET_IF_EMPTY(TEST_PICKUP_REGEX "$ENV{TEST_PICKUP_REGEX}")
-  GET_FILENAME_COMPONENT(_category ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+
+  SET(extra_macro_args ${ARGN})
+  LIST(LENGTH extra_macro_args num_extra_args)
+  IF (${num_extra_args} GREATER 0)
+    LIST(GET extra_macro_args 0 optional_arg)
+    SET(_category ${optional_arg})
+  ELSE()
+    GET_FILENAME_COMPONENT(_category ${CMAKE_CURRENT_SOURCE_DIR} NAME)
+  ENDIF()
```

Comment:
Would you mind shrinking this down to five lines as follows (untested):
```
IF("${ARGN}" STREQUAL "")
  GET_FILENAME_COMPONENT(_category ${CMAKE_CURRENT_SOURCE_DIR} NAME)
ELSE()
  SET(_category "${ARGN}")
ENDIF()
```

Reply:
I like the change - sorry for the nitpick

Reply:
I'll try it in a minute.

Reply:
Yup works!

Reply:
Thanks!